src/main/java/com/yourssu/roomescape/reservation/ReservationService.java

```diff
@@ -1,30 +1,147 @@
+                theme
+        );
+
+        Reservation savedReservation = reservationRepository.save(reservation);
+
+        return new ReservationResponse(
+                savedReservation.getId(),
+                reservationRequest.getName(),
+                theme.getName(),
+                savedReservation.getDate(),
+                time.getValue()
+        );
     }
 
+    @Transactional
+    public ReservationResponse save(ReservationRequest reservationRequest, LoginMember loginMember) {
+        String name = reservationRequest.getName() != null ?
+                reservationRequest.getName() : loginMember.getName();
+
+        Time time = timeRepository.findById(reservationRequest.getTime())
+                .orElseThrow(() -> new RuntimeException("Time not found"));
+        Theme theme = themeRepository.findById(reservationRequest.getTheme())
+                .orElseThrow(() -> new RuntimeException("Theme not found"));
+
+        Member member = null;
+        if (loginMember != null) {
+            member = memberRepository.findById(loginMember.getId())
+                    .orElse(null);
+        }
+
+        Reservation reservation = new Reservation(
+                name,
+                reservationRequest.getDate(),
+                time,
+                theme,
+                member
+        );
+
+        Reservation savedReservation = reservationRepository.save(reservation);
+
+        return new ReservationResponse(
+                savedReservation.getId(),
+                name,
+                theme.getName(),
+                savedReservation.getDate(),
+                time.getValue()
+        );
+    }
+
+    @Transactional
     public void deleteById(Long id) {
-        reservationDao.deleteById(id);
+        reservationRepository.deleteById(id);
     }
 
     public List<ReservationResponse> findAll() {
-        return reservationDao.findAll().stream()
-                .map(it -> new ReservationResponse(it.getId(), it.getName(), it.getTheme().getName(), it.getDate(), it.getTime().getValue()))
+        return reservationRepository.findAll().stream()
+                .map(it -> new ReservationResponse(
+                        it.getId(),
+                        it.getName(),
+                        it.getTheme().getName(),
+                        it.getDate(),
+                        it.getTime().getValue()
+                ))
                 .toList();
     }
-}
+
+    public List<MyReservationResponse> findMyReservations(Member member) {
+        // 예약 목록 가져오기
+        List<MyReservationResponse> reservations = reservationRepository.findByMember(member).stream()
+                .map(it -> new MyReservationResponse(
+                        it.getId(),
+                        it.getTheme().getName(),
+                        it.getDate(),
+                        it.getTime().getValue(),
+                        "예약"
+                ))
+                .collect(Collectors.toList());
```

Comment:
```suggestion
                ))
                .toList();
```

이렇게 할 수도 있어요. 두 방식의 차이가 뭘까요?

Reply:
버전 차이도 있고, 불변 리스트를 반환하는지 아닌지의 차이가 있는 것 같아요. toList()로 하는게 더 깔끔하고 불변이기도 하니 제안해주신 방법으로 수정할게요!